table/block_based_table_builder.cc

```diff
@@ -527,13 +527,13 @@ void BlockBasedTableBuilder::WriteBlock(const Slice& raw_block_contents,
     RecordTick(r->ioptions.statistics, NUMBER_BLOCK_NOT_COMPRESSED);
     type = kNoCompression;
     block_contents = raw_block_contents;
-  } else if (type != kNoCompression &&
-             ShouldReportDetailedTime(r->ioptions.env,
-                                      r->ioptions.statistics)) {
-    MeasureTime(r->ioptions.statistics, COMPRESSION_TIMES_NANOS,
-                timer.ElapsedNanos());
-    MeasureTime(r->ioptions.statistics, BYTES_COMPRESSED,
-                raw_block_contents.size());
+  } else if (type != kNoCompression) {
+    if (ShouldReportDetailedTime(r->ioptions.env, r->ioptions.statistics)) {
+      MeasureTime(r->ioptions.statistics, COMPRESSION_TIMES_NANOS,
+                  timer.ElapsedNanos());
+      MeasureTime(r->ioptions.statistics, BYTES_COMPRESSED,
+                  raw_block_contents.size());
+    }
     RecordTick(r->ioptions.statistics, NUMBER_BLOCK_COMPRESSED);
   }
 
```

Comment:
Can we add a stat to also measure the block_contents size? That would give the compression ratio stats that Mark was asking for.

Reply:
Add it by yourself if you want:)
I'm just doing a bug fixing.

Reply:
By the way, please note this is a histogram. I'm not sure how you can get compression ratio from a histogram and another counter.